exp/controllers/azuremachinepoolmachine_controller.go

```diff
@@ -253,6 +292,8 @@ func (ampmr *AzureMachinePoolMachineController) reconcileNormal(ctx context.Cont
 		return reconcile.Result{}, nil
 	}
 
+	machineScope.UpdateDeleteMachineAnnotation()
```

Comment:
Similar to above, not sure this is the best place to call this. In this case it would make sense to do this closer to where the annotation is being used, when selecting machines for delete.

Reply:
When we're selecting machines for delete, we're already in the AMP controller, which would require us to pull each of the owner Machines for each AMPM. I figured it would be easier to do so here as we already have access to the AMPM and its owner Machine. I think we could move it to `Close()` instead, though since that's when it gets patched.